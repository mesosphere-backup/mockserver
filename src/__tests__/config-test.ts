import log from "npmlog";
import getConfig, {
  getPort,
  getProxyHostPort,
  getProxyHost,
  getProxyPort
} from "../config";

describe("Config", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe("#getPort", () => {
    it("returns a default port if no port is configured", () => {
      expect(getPort()).toBe(3000);
    });

    it("returns the configured port", () => {
      process.env.PORT = "4002";
      expect(getPort()).toBe(4002);
      process.env.PORT = undefined;
    });
  });

  describe("#getProxyHostPort", () => {
    afterEach(() => {
      delete process.env.PROXY_HOST_PORT;
    });

    it("returns a string if set correctly", () => {
      process.env.PROXY_HOST_PORT = "my-application.com:4310";
      const hostPort = getProxyHostPort();
      expect(hostPort).toBe("my-application.com:4310");
    });

    it("throws an error if nothing is passed as environment variable", () => {
      expect(() => {
        getProxyHostPort();
      }).toThrowErrorMatchingSnapshot();
    });

    it("throws an error if the protocol is included in the environment variable", () => {
      process.env.PROXY_HOST_PORT = "http://my-application.com:4310";
      expect(() => {
        getProxyHostPort();
      }).toThrowErrorMatchingSnapshot();
    });

    it("shows a warning if no port is configured", () => {
      process.env.PROXY_HOST_PORT = "my-application.com";
      const hostPort = getProxyHostPort();
      expect(hostPort).toBe("my-application.com:80");
      expect(log.warn).toHaveBeenCalled();
    });
  });

  describe("#getProxyHost", () => {
    it("extracts the proxy host", () => {
      expect(getProxyHost("google.de:80")).toBe("google.de");
    });
  });

  describe("#getProxyPort", () => {
    it("extracts the proxy port", () => {
      expect(getProxyPort("google.de:80")).toBe(80);
    });
  });

  describe("#getConfig", () => {
    it("returns a complete configuration", () => {
      process.env.PROXY_HOST_PORT = "my-application.com:1234";
      process.env.PORT = "4002";

      expect(getConfig()).toEqual({
        port: 4002,
        proxyPort: 1234,
        proxyHost: "my-application.com"
      });

      process.env.PORT = undefined;
    });
  });
});
