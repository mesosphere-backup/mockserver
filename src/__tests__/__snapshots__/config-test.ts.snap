// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config #getMockSearchExpression fails without the environment variable 1`] = `
"mockserver needs the MOCK_SEARCH_EXPRESSION environment variable to be set.
    It is used to determine where to look for the mocks and it's expected to be a glob expression."
`;

exports[`Config #getProxyHostPort throws an error if nothing is passed as environment variable 1`] = `
"mockserver needs the PROXY_HOST_PORT environment variable to be set.
It is used to determine where to send the requests no mock matches against to.
E.g. PROXY_HOST_PORT='my-service.com:2342'"
`;

exports[`Config #getProxyHostPort throws an error if the protocol is included in the environment variable 1`] = `"Please remove the protocol from your PROXY_HOST_PORT"`;
